buildlib/jucx-publish.yml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  target: publish
+  temp_cfg: $(System.DefaultWorkingDirectory)/bindings/java/src/main/native/build-java/tmp-settings.xml
+  gpg_dir: $(System.DefaultWorkingDirectory)/bindings/java/src/main/native/build-java/gpg
+
+steps:
+  - bash: |
+      set -eE
+      {
+        echo -e "<settings><servers><server>"
+        echo -e "<id>ossrh</id><username>\${env.SONATYPE_USERNAME}</username>"
+        echo -e "<password>\${env.SONATYPE_PASSWORD}</password>"
+        echo -e "</server></servers></settings>"
+      } > ${{ parameters.temp_cfg }}
+    displayName: Generate temporary config
+
+  - task: DownloadSecureFile@1
+    displayName: Download Secure file
+    inputs:
+      secureFile: sparkucx-secret.gpg
+    name: privateKey
+
+  - task: DownloadSecureFile@1
+    displayName: Download Secure file
+    inputs:
+      secureFile: sparkucx-public.gpg
+    name: publicKey
+
+  - bash: |
+      set -eE
+      TAG=`git describe --tags`
+      make -C bindings/java/src/main/native/ set-version JUCX_VERSION=${TAG:1}
+    displayName: "Set JUCX release version"
+    condition: and(succeeded(), eq('${{ parameters.target }}', 'publish'))
```

Comment:
what is this doing?

Reply:
There's 2 option for `${{ parameters.target }}`: `publish` and `publish-snapshot`. Need to set version only when target is `publish` - when we publish release

Reply:
can't it be part of makefile same way you did in publish-shanpshot?

Reply:
It can, but `make set-version` is a general function. Adding git to Makefile would require to have it + would be impossible to publish from non-git distribution (e.g. src rpm or tarball).

Reply:
can publish release step do 
`TAG=git describe --tags`
`make publish JUCX_VERSION=${TAG:1}`